astropy/units/physical.py

```diff
@@ -1,133 +1,461 @@
+    (astrophys.electron * si.m ** -2 * si.s ** -1, "electron flux"),
+    (si.kg / si.m ** 2, "surface mass density"),
+    (si.W / si.m ** 2 / si.sr, "radiance"),
+    (si.J / si.mol, "chemical potential"),
+    (si.kg / si.m, "linear density"),
+    (si.H ** -1, "magnetic reluctance"),
+    (si.W / si.K, "thermal conductance"),
+    (si.K / si.W, "thermal resistance"),
+    (si.K * si.m / si.W, "thermal resistivity"),
+    (si.N / si.s, "yank"),
+    (si.S * si.m ** 2 / si.mol, "molar conductivity"),
+    (si.m ** 2 / si.V / si.s, "electrical mobility"),
+    (si.lumen / si.W, "luminous efficacy"),
+    (si.m ** 2 / si.kg, {"opacity", "mass attenuation coefficient"}),
+    (si.kg * si.m ** -2 * si.s ** -1, {"momentum density", "mass flux"}),
+    (si.m ** -3, "number density"),
+    (si.m ** -2 * si.s ** -1, "particle flux"),
+]
 
 _physical_unit_mapping = {}
 _unit_physical_mapping = {}
 
 
+def _replace_temperatures_with_kelvin(unit):
+    """
+    If a unit contains a temperature unit besides kelvin, then replace
+    that unit with kelvin.
+
+    Temperatures cannot be converted directly between K, °F, °C, and
+    °Ra, in particular since there would be different conversions for
+    T and ΔT.  However, each of these temperatures each represents the
+    physical type.  Replacing the different temperature units with
+    kelvin allows the physical type to be treated consistently.
+    """
+    physical_type_id = unit._get_physical_type_id()
+
+    physical_type_id_components = []
+    substitution_was_made = False
+
+    for base, power in physical_type_id:
+        if base in ["deg_F", "deg_C", "deg_R"]:
+            base = "K"
+            substitution_was_made = True
+        physical_type_id_components.append((base, power))
+
+    if substitution_was_made:
+        return core.Unit._from_physical_type_id(tuple(physical_type_id_components))
+    else:
+        return unit
+
+
+def _standardize_physical_type_names(physical_type_input):
+    """
+    Convert a string or `set` of strings into a `set` containing
+    string representations of physical types.
+
+    The strings provided in ``physical_type_input`` can each contain
+    multiple physical types that are separated by a regular slash.
+    Underscores are treated as spaces so that variable names could
+    be identical to physical type names.
+    """
+
+    if isinstance(physical_type_input, str):
+        physical_type_input = {physical_type_input}
+
+    standardized_physical_types = set()
+
+    for ptype_input in physical_type_input:
+        if not isinstance(ptype_input, str):
+            raise ValueError(f"expecting a string, but got {ptype_input}")
+        input_set = set(ptype_input.split("/"))
+        processed_set = {s.strip().replace("_", " ") for s in input_set}
+        standardized_physical_types |= processed_set
+
+    return standardized_physical_types
+
+
+def _identify_unit_from_unit_or_physical_type(obj):
```

Comment:
I think I'd just make it a method on the class, since it is only used there, but obviously no big deal.